src/core/bptree_set.h

```diff
@@ -484,6 +492,46 @@ size_t BPTree<T, Policy>::DeleteRangeByRank(uint32_t start, uint32_t end,
   return deleted;
 }
 
+template <typename T, typename Policy>
+size_t BPTree<T, Policy>::DeleteRange(KeyT first, KeyT last, std::function<void(KeyT)> cb,
+                                      bool inclusive_first, bool inclusive_last) {
```

Comment:
I thought this could be just turned into a call to DeleteRangeByRank... But its actually less effective to additionally search for the right boundary if it'll be reached by iteration either way, so never mind